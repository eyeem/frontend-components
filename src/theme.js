import { BREAKPOINTS } from './styleUtils';

export const colors = {
  whites: {
    white: '#ffffff',
    pearlWhite: '#fafafa',
    transparent: 'rgba(0,0,0,0)'
  },

  greys: {
    grey1: '#666666',
    grey2: '#888888',
    grey3: '#999999',
    grey4: '#cccccc',
    grey5: '#e5e5e5',
    grey6: '#f2f2f2',
    grey7: '#f5f5f5'
  },

  blacks: {
    black1: '#000000',
    black2: '#111111',
    black3: '#1d1d1d',
    black4: '#292929',
    black5: '#333333'
  },

  lightColors: {
    lightRed: '#fff5f5',
    lightBlue: '#e7f0fb'
  },

  solidColors: {
    green: '#22cc88',
    greenv2: '#48B307',
    greenHover: '#4ed69f',
    greenActive: '#1ba36c',
    orange: '#ffaa33',
    orangeHover: '#ffbb5b',
    orangeActive: '#cc8828',
    red: '#ff3333',
    redHover: '#ff5b5b',
    redActive: '#cc2828',
    redv2: '#EA3223'
  },

  foreigns: {
    tumblr: '#34465d',
    twitter: '#55acee',
    facebook1: '#3a5795',
    facebook2: '#3b5998',
    facebook2Hover: '#4c70ba',
    facebook2Disabled: '#96abd6',
    paypal: '#009cde',
    paypalActive: '#007CB1',
    paypalDisabled: '#7FCEEF',
    paypalHover: '#008CC8'
  },

  uncategorized: {
    unnamed1: '#b2b2b2',
    unnamed3: 'rgba(255,255,255,.2)',
    unnamed7: 'rgba(51,51,51,.6)',
    unnamed8: 'rgba(0,0,0,.8)',
    unnamed9: 'rgba(0,0,0,.6)',
    unnamed10: 'rgba(0,0,0,.4)',
    unnamed12: 'rgba(255,255,255,.3)',
    unnamed13: 'rgba(255,255,255,.1)',
    unnamed15: 'rgba(0,0,0,.1)',
    unnamed16: 'rgba(0,0,0,.2)',
    unnamed17: 'rgba(0,0,0,.7)',
    unnamed24: '#ff6666',
    unnamed28: 'rgba(255,255,255,.6)',
    unnamed30: 'rgba(255,255,255,.5)',
    unnamed31: '#5555ff',
    unnamed39: 'rgba(0,0,0,.5)'
  }
};

export const borderRadius = '2px';

/* NOTE: styled-system
         https://styled-system.com/#space-theming
         https://styled-system.com/#responsive-styles
*/
export const space = [0, 4, 8, 16, 32, 64, 128, 256, 512];
export const breakpoints = {
  sm: `${BREAKPOINTS.small}px`,
  md: `${BREAKPOINTS.medium}px`,
  lg: `${BREAKPOINTS.large}px`,
  xl: `${BREAKPOINTS.xlarge}px`
};
