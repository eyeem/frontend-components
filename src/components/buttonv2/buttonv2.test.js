import React from 'react';
import { ThemeProvider } from 'styled-components';
import { render, getByText, cleanup, fireEvent } from '@testing-library/react';

import { colors } from '../../theme';
import Buttonv2 from './buttonv2';

const wrapWithTheme = component => (
  <ThemeProvider theme={{ mode: 'dark' }}>{component}</ThemeProvider>
);

afterEach(cleanup);

test('it renders default button', () => {
  const { container } = render(<Buttonv2>Hello World</Buttonv2>);
  const hasRenderedChildText = getByText(container, 'Hello World');
  const component = container.firstChild;
  expect(hasRenderedChildText).toBeTruthy();
  expect(component).toMatchSnapshot();
});

test('it accepts property disabled', () => {
  const { container } = render(<Buttonv2 disabled>Disabled Button</Buttonv2>);
  const button = container.firstChild;
  expect(button).toHaveProperty('disabled');
  expect(button).toMatchSnapshot();
});

test('it accepts property fullWidth', () => {
  const { container } = render(<Buttonv2 fullWidth>FullWidth Button</Buttonv2>);
  const button = container.firstChild;
  expect(button).toHaveStyleRule('width', '100%');
  expect(button).toMatchSnapshot();
});

test('it accepts property icon=checkmark', () => {
  const { container } = render(
    <Buttonv2 icon="checkmark">Icon Button</Buttonv2>
  );
  const button = container.firstChild;
  expect(button.firstChild.nodeName).toBe('svg');
  expect(button).toMatchSnapshot();
});

test('it renders black background in dark mode with outline fill set', () => {
  const { container } = render(
    wrapWithTheme(<Buttonv2 fill="outline">Color Button</Buttonv2>)
  );

  const component = container.firstChild;
  const backgroundColor = colors.blacks.black1;
  expect(component).toHaveStyleRule('background-color', backgroundColor);
  expect(component).toMatchSnapshot();
});

test('onClick handler beeing called', () => {
  const onClickHandler = jest.fn();
  const { container } = render(
    <Buttonv2 onClick={onClickHandler}>Click Button</Buttonv2>
  );
  const button = container.firstChild;
  fireEvent.click(button);
  expect(onClickHandler).toHaveBeenCalledTimes(1);
});
