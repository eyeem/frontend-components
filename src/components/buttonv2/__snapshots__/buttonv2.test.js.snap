// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it accepts property disabled 1`] = `
.c0 {
  position: relative;
  display: inline-block;
  max-width: 100%;
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
  width: auto;
  font-size: 14px;
  line-height: 20px;
  padding: 12px 20px;
  max-height: 48px;
  border-width: 1px;
  border-style: solid;
  vertical-align: top;
  -webkit-text-decoration: none;
  text-decoration: none;
  text-align: center;
  text-shadow: none;
  cursor: pointer;
  -webkit-transition: background-color 0.2s ease-in-out,border 0.2s ease-out, color 0.2s ease-out,fill 0.2s ease-out;
  transition: background-color 0.2s ease-in-out,border 0.2s ease-out, color 0.2s ease-out,fill 0.2s ease-out;
  pointer-events: all;
  font-weight: 300;
  font-family: "Sailec Regular",sans-serif;
}

.c0:focus {
  outline: none;
}

<button
  class="c0"
  disabled=""
>
  Disabled Button
</button>
`;

exports[`it accepts property fullWidth 1`] = `
.c0 {
  position: relative;
  display: inline-block;
  max-width: 100%;
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
  width: 100%;
  font-size: 14px;
  line-height: 20px;
  padding: 12px 20px;
  max-height: 48px;
  border-width: 1px;
  border-style: solid;
  vertical-align: top;
  -webkit-text-decoration: none;
  text-decoration: none;
  text-align: center;
  text-shadow: none;
  cursor: pointer;
  -webkit-transition: background-color 0.2s ease-in-out,border 0.2s ease-out, color 0.2s ease-out,fill 0.2s ease-out;
  transition: background-color 0.2s ease-in-out,border 0.2s ease-out, color 0.2s ease-out,fill 0.2s ease-out;
  pointer-events: all;
  font-weight: 300;
  font-family: "Sailec Regular",sans-serif;
}

.c0:focus {
  outline: none;
}

<button
  class="c0"
>
  FullWidth Button
</button>
`;

exports[`it accepts property icon=checkmark 1`] = `
.c0 {
  position: relative;
  display: inline-block;
  max-width: 100%;
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
  width: auto;
  font-size: 14px;
  line-height: 20px;
  padding: 12px 20px;
  max-height: 48px;
  border-width: 1px;
  border-style: solid;
  vertical-align: top;
  -webkit-text-decoration: none;
  text-decoration: none;
  text-align: center;
  text-shadow: none;
  cursor: pointer;
  -webkit-transition: background-color 0.2s ease-in-out,border 0.2s ease-out, color 0.2s ease-out,fill 0.2s ease-out;
  transition: background-color 0.2s ease-in-out,border 0.2s ease-out, color 0.2s ease-out,fill 0.2s ease-out;
  pointer-events: all;
  font-weight: 300;
  font-family: "Sailec Regular",sans-serif;
}

.c0:focus {
  outline: none;
}

.c1 {
  margin-right: 8px;
  margin-bottom: -2px;
}

<button
  class="c0"
>
  <svg
    class="c1"
    height="16"
    viewBox="0 0 24 24"
    width="16"
  >
    <path
      d="M2.5 12l6.806 7L21.5 5"
      fill="none"
      stroke="#fff"
      stroke-width="2"
    />
  </svg>
  Icon Button
</button>
`;

exports[`it renders black background in dark mode with outline fill set 1`] = `
.c0 {
  position: relative;
  display: inline-block;
  max-width: 100%;
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
  width: auto;
  font-size: 14px;
  line-height: 20px;
  padding: 12px 20px;
  max-height: 48px;
  border-color: #ffffff;
  border-width: 1px;
  border-style: solid;
  background-color: #000000;
  color: #ffffff;
  fill: #ffffff;
  vertical-align: top;
  -webkit-text-decoration: none;
  text-decoration: none;
  text-align: center;
  text-shadow: none;
  cursor: pointer;
  -webkit-transition: background-color 0.2s ease-in-out,border 0.2s ease-out, color 0.2s ease-out,fill 0.2s ease-out;
  transition: background-color 0.2s ease-in-out,border 0.2s ease-out, color 0.2s ease-out,fill 0.2s ease-out;
  pointer-events: all;
  font-weight: 300;
  font-family: "Sailec Regular",sans-serif;
}

.c0:focus,
.c0:hover {
  color: #cccccc;
  fill: #cccccc;
  background-color: #000000;
  border-color: #cccccc;
}

.c0:active {
  background-color: #000000;
  border-color: #888888;
  fill: #888888;
  color: #888888;
}

.c0:focus {
  outline: none;
}

.c0:disabled {
  background-color: #000000;
  border-color: #666666;
  fill: #666666;
  color: #666666;
}

<button
  class="c0"
>
  Color Button
</button>
`;

exports[`it renders default button 1`] = `
.c0 {
  position: relative;
  display: inline-block;
  max-width: 100%;
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
  width: auto;
  font-size: 14px;
  line-height: 20px;
  padding: 12px 20px;
  max-height: 48px;
  border-width: 1px;
  border-style: solid;
  vertical-align: top;
  -webkit-text-decoration: none;
  text-decoration: none;
  text-align: center;
  text-shadow: none;
  cursor: pointer;
  -webkit-transition: background-color 0.2s ease-in-out,border 0.2s ease-out, color 0.2s ease-out,fill 0.2s ease-out;
  transition: background-color 0.2s ease-in-out,border 0.2s ease-out, color 0.2s ease-out,fill 0.2s ease-out;
  pointer-events: all;
  font-weight: 300;
  font-family: "Sailec Regular",sans-serif;
}

.c0:focus {
  outline: none;
}

<button
  class="c0"
>
  Hello World
</button>
`;
